internal/controller/deployment/controller_finalize.go

```diff
@@ -87,3 +129,48 @@ func (r *Reconciler) finalize(ctx context.Context, old, deployment *choreov1.Dep
 	}
 	return ctrl.Result{}, nil
 }
+
+func (r *Reconciler) cleanupEndpoints(ctx context.Context, deployment *choreov1.Deployment) error {
+	logger := log.FromContext(ctx).WithValues("deployment", deployment.Name)
+	logger.Info("Cleaning up the endpoints associated with the deployment")
+
+	// List all the endpoints associated with the deployment
+	endpointList := &choreov1.EndpointList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(deployment.Namespace),
+		client.MatchingLabels{
+			labels.LabelKeyDeploymentName: deployment.Name,
+		},
+	}
+
+	if err := r.List(ctx, endpointList, listOpts...); err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			logger.Info("No endpoints associated with the environment")
+			return nil
+		}
```

Comment:
Do we need this? AFAIR, the listing does not return 404. It will return an empty slice.

Reply:
yeah you are right. I tested the behaviour it is returning an empty slice